pre_commit/languages/rust.py

```diff
@@ -84,23 +164,29 @@ def install_environment(
         packages_to_install: set[tuple[str, ...]] = {('--path', '.')}
         for cli_dep in cli_deps:
             cli_dep = cli_dep[len('cli:'):]
-            package, _, version = cli_dep.partition(':')
-            if version != '':
-                packages_to_install.add((package, '--version', version))
+            package, _, crate_version = cli_dep.partition(':')
+            if crate_version != '':
+                packages_to_install.add((package, '--version', crate_version))
             else:
                 packages_to_install.add((package,))
 
-        for args in packages_to_install:
-            cmd_output_b(
-                'cargo', 'install', '--bins', '--root', directory, *args,
-                cwd=prefix.prefix_dir,
-            )
+        with in_env(prefix, version):
+            toolchain = TOOLCHAIN_FROM_VERSION.get(version, version)
```

Comment:
I don't think `TOOLCHAIN_FROM_VERSION` is all that useful -- this condition here is just `if version != 'system':` rather than needing to go through this mapping

Reply:
I only added the mapping to have a single source of truth for mapping `C.DEFAULT` to `stable`. I think I incorporated the changes in https://github.com/pre-commit/pre-commit/pull/2534/commits/8d545adedbcf33c731a995f7b2fe22fa1be2bef4